### PR TITLE
Add support for BGPMon over IPv6 for T2 chassis

### PR DESCRIPTION
**Pull request by tjchadaga**
### Description of PR

Summary:
Tests for IPv6-based bgpmon for T2 chassis

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012
- [x] 202205

#### What is the motivation for this PR?
Add new test cases for IPv6-based bgpmon session

#### How did you do it?
Modified tests to use IPv6 loopback4096 address as source and initiate v6 BGPMon session to exabgp on PTF

#### How did you verify/test it?
Tested on T2 chassis

### CODE REVIEW

File: tests/bgp/test_bgpmon.py [hunk 1]
```diff
@@ -143,10 +109,12 @@ def build_syn_pkt(local_addr, peer_addr):
 
 
 def test_bgpmon(dut_with_default_route, localhost, enum_rand_one_frontend_asic_index,
-                common_setup_teardown, set_timeout_for_bgpmon, ptfadapter, ptfhost):
+                common_setup_teardown, set_timeout_for_bgpmon, ptfadapter, ptfhost, tbinfo):
     """
     Add a bgp monitor on ptf and verify that DUT is attempting to establish connection to it
     """
+    if tbinfo['topo']['type'] == 't2':
```
Comment (arlakshm) [inline tests/bgp/test_bgpmon.py:116]: iso of skipping here, 
```suggestion
    if tbinfo['topo']['type'] == 't2':
```can we add conditional marker to skip these test?

File: tests/bgp/test_bgpmon_v6.py [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+        ptfhost.shell("ip -6 route del %s dev %s" % (local_addr + "/128", ptf_interface))
+        ptfhost.shell("ip -6 neigh del %s lladdr %s dev %s" % (local_addr, duthost.facts["router_mac"], ptf_interface))
+        ptfhost.shell("ip -6 addr del %s dev %s" % (peer_addr + "/128", ptf_interface))
+        ptfhost.shell("ifconfig %s hw ether %s" % (ptf_interface, original_mac))
+
+
+def test_bgpmon_no_ipv6_resolve_via_default(dut_with_default_route, enum_rand_one_frontend_asic_index,
+                                            common_v6_setup_teardown, ptfadapter):
+    """
+    Verify no syn for BGP is sent when 'ip nht resolve-via-default' is disabled.
```
Comment (arlakshm) [inline tests/bgp/test_bgpmon_v6.py:170]: nit: ipv6

**Review by arlakshm — COMMENTED** [review]
lgtm, couple of minor comments

**Review by arlakshm — APPROVED** [review]

**Comment by tjchadaga** [comment]
> lgtm, couple of minor comments

@arlakshm - addressed all comments. Please check

**Review by arlakshm — APPROVED** [review]

**Comment by wangxin** [comment]
@tjchadaga This is to add new tests. Do we really need it in 202205 branch?

**Comment by tjchadaga** [comment]
> @tjchadaga This is to add new tests. Do we really need it in 202205 branch?

@wangxin - in this PR, we are skipping the current bgpmon tests on T2 and would be running ipv6-based bgpmon tests instead. So these new tests are required to validate the feature on chassis. They will not run on any T0/T1 devices.

**Comment by tjchadaga** [comment]
@wangxin - could you please help cherry-pick this to 202205?

**Pull request merged**